src/libraries/System.Net.Http/src/System/Net/Http/HttpClientHandler.AnyMobile.cs

```diff
@@ -0,0 +1,423 @@
+        }
+
+        [UnsupportedOSPlatform("android")]
+        [UnsupportedOSPlatform("browser")]
+        [UnsupportedOSPlatform("ios")]
+        [UnsupportedOSPlatform("tvos")]
+        [UnsupportedOSPlatform("maccatalyst")]
+        public SslProtocols SslProtocols
+        {
+            get => throw new PlatformNotSupportedException();
+            set => throw new PlatformNotSupportedException();
+        }
+
+        [UnsupportedOSPlatform("android")]
+        [UnsupportedOSPlatform("ios")]
+        [UnsupportedOSPlatform("tvos")]
+        [UnsupportedOSPlatform("maccatalyst")]
+        public IDictionary<string, object?> Properties => throw new PlatformNotSupportedException();
+
+        //
+        // Attributes are commented out due to https://github.com/dotnet/arcade/issues/7585
+        // API compat will fail until this is fixed
+        //
+        //[UnsupportedOSPlatform("android")]
+        [UnsupportedOSPlatform("browser")]
+        //[UnsupportedOSPlatform("ios")]
+        //[UnsupportedOSPlatform("tvos")]
+        //[UnsupportedOSPlatform("maccatalyst")]
+        protected internal override HttpResponseMessage Send(HttpRequestMessage request,
+            CancellationToken cancellationToken)
+        {
+            throw new PlatformNotSupportedException();
+        }
+
+        protected internal override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request,
+            CancellationToken cancellationToken)
+        {
+            if (DiagnosticsHandler.IsGloballyEnabled() && _diagnosticsHandler != null)
+            {
+                return _diagnosticsHandler!.SendAsync(request, cancellationToken);
+            }
+
+            if (IsSocketHandler)
+            {
+                return _socketHandler!.SendAsync(request, cancellationToken);
+            }
+            else
+            {
+                return _underlyingHandler!.SendAsync(request, cancellationToken);
+            }
+        }
+
+        [UnsupportedOSPlatform("android")]
+        [UnsupportedOSPlatform("browser")]
+        [UnsupportedOSPlatform("ios")]
+        [UnsupportedOSPlatform("tvos")]
+        [UnsupportedOSPlatform("maccatalyst")]
+        public static Func<HttpRequestMessage, X509Certificate2?, X509Chain?, SslPolicyErrors, bool> DangerousAcceptAnyServerCertificateValidator =>
+            throw new PlatformNotSupportedException();
+
+        private void CheckDisposed()
+        {
+            if (_disposed)
+            {
+                throw new ObjectDisposedException(GetType().ToString());
+            }
+        }
+
+        private object InvokeNativeHandlerMethod(string name, params object?[] parameters)
+        {
+            return _underlyingHandler!.GetType()!.GetMethod(name)!.Invoke(_underlyingHandler, parameters)!;
+        }
+
+        private static bool IsSocketHandler => !IsNativeHandlerEnabled();
+
+        private static bool IsNativeHandlerEnabled() => RuntimeSettingParser.QueryRuntimeSettingSwitch(
```

Comment:
Do you need both?

Reply:
Yeah, it's kind of weird.  I lead with `IsSocketHandler` in all of the checks, but we have `UseNativeHttpHandler` coming from msbuild.  I guess I can flip the condition.